### PR TITLE
Add PyUpgrade Pre-Commit Hook for support python 3.7 and above

### PR DESCRIPTION
**Pull request by marload**
Removed format method and added f-string.

### CODE REVIEW

**Review by marctorsoc — APPROVED** [review]
Approved, thanks! if you can, could update the ones using % so we have everything with the same format? :)

**Comment by cclauss** [comment]
This process can largely be automated using [pyupgrade](https://github.com/asottile/pyupgrade) but we will need to drop [support for legacy Python](https://github.com/hyperopt/hyperopt/blob/master/setup.py#L33) to use f-strings because they are only available on Python >= 3.6.

**Comment by marload** [comment]
@cclauss 
We internally decided to support HyperOpt in Python 3.7 and above. Using a pyupgrade looks good. :smile:

**Comment by marload** [comment]
@marctorsoc @maxpumperla @cclauss 
I added a `pyupgrade --py37-plus` to the pre-commit. So, This PR title was changed a "Add PyUpgrade Pre-Commit Hook for support python 3.7 and above" 😄

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,4 +3,6 @@ repos:
     rev: stable
     hooks:
     - id: black
-      args: [--check]
\ No newline at end of file
+      args: [--check]
+    - id: pyupgrade
+      args: [--py37-plus]
```
Comment (cclauss) [inline .pre-commit-config.yaml:8]: Add one and only one carriage return to the end of the file so GitHub removes the ⛔ .

File: docs/autogen.py [hunk 2]
```diff
@@ -1,9 +1,6 @@
-# -*- coding: utf-8 -*-
 # This file has been taken from Keras' `docs` module found here:
 # https://github.com/keras-team/keras/blob/master/docs/autogen.py
 #
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (cclauss) [inline docs/autogen.py:6]: Lines 10 thru 14 can also be removed.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (marload) [reply .pre-commit-config.yaml]: complete :smile:

File: docs/autogen.py (hunk 2 above)
Comment (marload) [reply docs/autogen.py:6]: complete :smile:

**Review by cclauss — APPROVED** [review]
LGTM!

**Comment by marctorsoc** [comment]
Got late to this one, thanks everyone!

**Pull request merged**